docs/arch/adr-003-versioning-branching-strategy.md

```diff
@@ -0,0 +1,108 @@
+
+## Common Terminology
+* Release version: refers to the numeric value of the release (eg. `1.0.1`, `0.21.0`) There are major, minor, and patch release versions that map to standard [semantic versioning practices](https://semver.org/#semantic-versioning-200).
+* Release phase: refers to the stability of the release (eg. beta, rc, final). 
+
+## Requirements
+The following are requirements that must be met for releasing dbt:
+
+1. We must have major, minor, and patch releases.
+1. We must have release phases that have the goal of garnering early feedback from the Community.
+1. We must have release phases that aim to identify bugs and test for stability.
+1. We must have release phases that are stable and of production quality.
+1. We must have the ability to isolate changes from different versions and phases to release.
+
+## Branching Strategy Options
+The 2 branching strategies we are considering using going forward are:
+	1. A new branch for each and every release
+	1. A new branch for each minor version which we release minor and patch versions from (current strategy)
+
+### New branch for every release
+#### Pros: 
+* Easy mapping between a release and a branch. Great for troubleshooting issues.
+* Less confusing and more straightforward for running a release on a branch. No question of what version is being released.
+#### Cons:
+* Breaks current adapters CI which needs to point to the exact `dbt-core` branch that it needs to test with. The branches would need to be updated in all adapter repos for each final and prerelease branch every time we have a new release branch for `dbt-core`. 
+* Confusing for backporting. We will need to keep creating new labels and making sure we remove the old labels because we don't want to accidently backport to the wrong branch.
+
+### Minor release branches (current strategy)
+#### Pros:
+* Adapter CI testing will continue to work with this strategy and makes it less complicated overall. Adapters and `dbt-core` need to be on the same minor version but can be on different patch versions. The current CI workflows verify that the changes slated for a new `1.1` patch are compatible across all repos by checking out the `1.1.latest` branch of each repo. The 1:1 mapping of adapter branches to `dbt-core` branches makes this easy.
+* Easy backporting. There is no need to figure out the exact branch that you need to backport to and accidently backporting to an older, stale branch.
+#### Cons:
+* Branches contain more than 1 release which is confusing. Without a 1:1 mapping between branch and release, you must rely on tags instead to know where in the branch the release was cut.
+* Releasing from the same branch multiple times can lead to confusion as to which version we are releasing and actually stops us from releasing the wrong version.
```

Comment:
> and actually stops us from releasing the wrong version.

I think I know what you're getting at, but the wording is confusing to me here. Maybe it should be a separate sentence.

Reply:
I see what you mean about the wording. I took another go at clarifying that

Reply:
the specifics make it much clearer. thanks!